crates/rpc_client/src/lib.rs

```diff
@@ -0,0 +1,134 @@
+// provide interfaces for all of the client functionality depended on by the existing Hardhat
+// Network logic, specifically
+// packages/hardhat-core/src/internal/hardhat-network/provider/fork/{ForkBlockchain,
+// ForkStateManager}.ts
+// and even more specifically, all of the methods shown in these excerpts:
+//
+//      const remote = await this._jsonRpcClient.getTransactionByHash(
+//    const remote = await this._jsonRpcClient.getTransactionReceipt(
+//      const remoteLogs = await this._jsonRpcClient.getLogs({
+//    const rpcBlock = await this._jsonRpcClient.getBlockByHash(blockHash, true);
+//    const rpcBlock = await this._jsonRpcClient.getBlockByNumber(
+//      const noncePromise = this._jsonRpcClient.getTransactionCount(
+//      const accountData = await this._jsonRpcClient.getAccountData(
+//    const remoteValue = await this._jsonRpcClient.getStorageAt(
+//
+// TODO: finish implementing get-tx-by-hash, then do others
+
+// TODO: Consider using a standard ethereum types library instead of reinventing the wheel here.
+#[derive(serde::Deserialize, serde::Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Transaction {
+    /*
+    TODO: stop saying everything is a String, ideally by deferring to some standard type rather
+    than coding any of this here, or at the very least look into what data types QUANTITY and DATA
+    should be, perhaps involving RLP decoding for numbers, and who knows what for DATA. more
+    specifically, conform to the data types below, excerpted from https://ethereum.org/en/developers/docs/apis/json-rpc/#eth_gettransactionbyhash
+
+    blockHash: DATA, 32 Bytes - hash of the block where this transaction was in. null when its pending.
+    blockNumber: QUANTITY - block number where this transaction was in. null when its pending.
+    from: DATA, 20 Bytes - address of the sender.
+    gas: QUANTITY - gas provided by the sender.
+    gasPrice: QUANTITY - gas price provided by the sender in Wei.
+    hash: DATA, 32 Bytes - hash of the transaction.
+    input: DATA - the data send along with the transaction.
+    nonce: QUANTITY - the number of transactions made by the sender prior to this one.
+    to: DATA, 20 Bytes - address of the receiver. null when its a contract creation transaction.
+    transactionIndex: QUANTITY - integer of the transactions index position in the block. null when its pending.
+    value: QUANTITY - value transferred in Wei.
+    v: QUANTITY - ECDSA recovery id
+    r: QUANTITY - ECDSA signature r
+    s: QUANTITY - ECDSA signature s
+    */
+    block_hash: String,
+    block_number: String,
+    from: String,
+    gas: String,
+    gas_price: String,
+    hash: String,
+    input: String,
+    nonce: String,
+    to: String,
+    transaction_index: String,
+    value: String,
+    v: String,
+    r: String,
+    s: String,
+}
+
+pub fn get_tx_by_hash(
+    url: String,
```

Comment:
Prefer `&str`, a string slice, as it can accept multiple types of data storage, not just a `String`

Reply:
Addressed in https://github.com/NomicFoundation/hardhat/pull/3383/commits/006e0837eea95f42b9db6c691f4a9088e4cbccc8